src/middlewared/middlewared/plugins/pool_/pool.py

```diff
@@ -680,6 +697,31 @@ async def do_create(self, job, data):
         self.middleware.send_event('pool.query', 'ADDED', id=pool_id, fields=pool)
         return pool
 
+    @private
+    async def validate_dedup_table_quota(self, data, verrors, schema='pool_create'):
+        dedup_table_quota = data.get('dedup_table_quota')
+        dedup_table_quota_value = data.get('dedup_table_quota_value')
+        if dedup_table_quota != 'CUSTOM' and dedup_table_quota_value is not None:
+            verrors.add(
+                f'{schema}.dedup_table_quota',
+                'You must set Deduplication Table Quota to CUSTOM to specify a value.',
+            )
+        elif dedup_table_quota == 'CUSTOM' and dedup_table_quota_value is None:
+            verrors.add(
+                f'{schema}.dedup_table_quota_value',
+                'This field is required when Deduplication Table Quota is set to CUSTOM.',
+            )
+
+        if verrors or 'dedup_table_quota' not in data:
```

Comment:
How can this be true?
```
if verrors or 'dedup_table_quota' not in data:
    return

if deduce_table_quota is None:
    return 'none'
```

The 2nd if branch can't be reached. It's also weird to still be checking if `dedup_table_quota not in data`

Reply:
The first can be true for the update case when default value will not be populated. So we explicitly check if the key has been set or not because if it has been specified and there are no validation errors up till this point - then the value is safe to consume